base/logger_file.go

```diff
@@ -140,6 +143,20 @@ func (lfc *FileLoggerConfig) init(level LogLevel, name string, logFilePath strin
 		lfc.CollationBufferSize = &bufferSize
 	}
 
+	if lfc.Rotation.RotatedLogsSizeLimit == 0 {
+		lfc.Rotation.RotatedLogsSizeLimit = defaultCumulativeMaxSizeBeforeDeletion
```

Comment:
Should probably move this into `initRotationConfig` now the field moved into the rotation struct.

Like we discussed, should validate that the value makes sense (i.e. -1 will break things)